metecho/api/models.py

```diff
@@ -1492,6 +1494,91 @@ def finalize_commit_changes(self, *, error=None, originating_user_id):
                 originating_user_id=originating_user_id,
             )
 
+    def queue_parse_datasets(self, *, user: User):
+        from .jobs import parse_datasets_job
+
+        self.currently_parsing_datasets = True
+        self.save()
+        self.notify_changed(originating_user_id=user.id)
+        parse_datasets_job.delay(org=self, user=user)
+
+    def finalize_parse_datasets(
+        self,
+        *,
+        error: Exception | None = None,
+        schema: dict[str, Any] | None = None,
+        dataset_errors: list[str] | None = None,
+        datasets: dict[str, Any] | None = None,
+        originating_user_id=None,
+    ):
+        # Notify about the boolean flag changing
+        self.currently_parsing_datasets = False
+        self.save()
+        self.notify_changed(originating_user_id=originating_user_id)
+
+        # The dataset information is not stored in the model, it's only pushed via WS
+        type_ = "SCRATCH_ORG_PARSE_DATASETS"
+        message = {
+            "schema": schema,
+            "dataset_errors": dataset_errors,
+            "datasets": datasets,
+        }
+        if error is not None:
+            type_ = "SCRATCH_ORG_PARSE_DATASETS_FAILED"
+            message = {
+                "dataset_errors": [
+                    str(
+                        _("Unable to parse dataset schema from Org: {}").format(
+                            str(error)
+                        )
+                    )
+                ],
+            }
```

Comment:
@jerivas Are you okay with this? I could add this on the FE if you prefer.

Reply:
That works for me 👍